caos.ansible_roles/roles/assert_process_running/tasks/assert-process-running-Linux.yaml

```diff
@@ -4,4 +4,16 @@
   shell: "pgrep -f {{ item }}"
   loop: "{{ processes_running }}"
 
+- name: assert processes are not running
+  shell: "pgrep -f {{ item }} || echo ''"
+  loop: "{{ processes_not_running }}"
+  register: process_result
+
+- name: fail if processes are running
+  assert:
+    that: "{{ item.stdout | length == 0}}"
+    msg: "process with PID {{ item.stdout }} should not be running"
+  loop: "{{ process_result.results }}"
+  when: '"results" in process_result'
```

Comment:
I think you can use `fail_when` and have a single task for this.

Reply:
Sorry I meant `failed_when`: https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#defining-failure

Reply:
That was my first approach (`failed_when: "process_result.stdout | length > 0"`), but then I couldn't have a custom message :(